src/core/chunked_stream.js

```diff
@@ -267,24 +245,489 @@ var ChunkedStream = (function ChunkedStreamClosure() {
+      this.buffer.start = 0;
+      this.buffer.startOffset = 0;
+      this.buffer.end = this.buffer.buffer.byteLength;
+      this.prepareBuffer = this.prepareBufferNop;
+      this.getByte = this.getByteFastContinuous(this.buffer.buffer, this.end);
+    }
+  };
+
+  ChunkedStreamContinuous.prototype.prepareBuffer = function
+      ChunkedStreamContinuous_prepareBufferContinuous(begin, end) {
+    if (!end) {
+      return;
+    }
+
+    // Checks if current buffer matches new [begin, end) parameters.
+    if (this.buffer.start <= begin && end <= this.buffer.end ||
+        end <= this.progressiveDataLength) {
+      return;
+    }
+
+    // Checks if there is initial block
+    if (this.initialChunk && this.initialChunk.start <= begin &&
+        end <= this.initialChunk.end) {
+      this.buffer = this.initialChunk;
+      return;
+    }
+
+    var chunkSize = this.chunkSize;
+    var beginChunk = Math.floor(begin / chunkSize);
+    var endChunk = Math.floor((end - 1) / chunkSize) + 1;
+    // Check if there are missing chunks.
+    for (var chunk = beginChunk; chunk < endChunk; ++chunk) {
+      if (!this.loadedChunks[chunk]) {
+        throw new MissingDataException(begin, end);
+      }
+    }
+
+    this.buffer.startOffset = 0;
+    this.buffer.start = begin;
+    this.buffer.end = end;
+  };
+
+  ChunkedStreamContinuous.prototype.onReceiveData = function
+      ChunkedStreamContinuous_onReceiveDataContinuous(begin, chunk) {
+    var chunkSize = this.chunkSize;
+    var data = new Uint8Array(chunk);
+
+    if (begin === 0 && data.byteLength % chunkSize !== 0) {
+      this.initialChunk = {
+        start:  0,
+        startOffset: 0,
+        end:    0,
+        buffer: this.buffer.buffer
+      };
+      this.buffer.buffer.set(data, this.initialChunk.end);
+      this.initialChunk.end += data.byteLength;
+
+      if (data.byteLength > chunkSize) {
+        data = data.subarray(0, data.byteLength -
+        data.byteLength % chunkSize);
+      } else {
+        return;
+      }
+    }
+
+    var end = begin + data.byteLength;
+    assert(begin % chunkSize === 0, 'Bad begin offset: ' + begin);
+
+    // Using this.length is inaccurate here since this.start can be moved
+    // See ChunkedStreamBase.moveStart()
+    var length = this.totalLength;
+    var beginChunk = Math.floor(begin / chunkSize);
+    var endChunk = Math.floor((end - 1) / chunkSize) + 1;
+
+    this.buffer.buffer.set(data, beginChunk * chunkSize);
+
+    assert(end % chunkSize === 0 || end === length,
+        'Bad end offset: ' + end);
+
+    var chunkEnd = chunkSize * beginChunk + data.byteLength;
+    if (this.loadedChunks[endChunk]) {
+      chunkEnd = this.loadedChunks[endChunk].end;
+    }
+
+    for (var curChunk = beginChunk; curChunk < endChunk; ++curChunk) {
+      if (!this.loadedChunks[curChunk]) {
+        this.loadedChunks[curChunk] = {
+          chunkOffset: chunkSize * curChunk,
+          start: 0,
+          startOffset: 0,
+          end:    chunkEnd,
```

Comment:
This line and the one below has too many spaces after `:`. Let's just use one as above and elsewhere in the codebase.